src/libraries/System.Runtime.Serialization.Formatters/ref/System.Runtime.Serialization.Formatters.cs

```diff
@@ -16,10 +16,12 @@ public abstract partial class Formatter : System.Runtime.Serialization.IFormatte
         public abstract System.Runtime.Serialization.StreamingContext Context { get; set; }
         public abstract System.Runtime.Serialization.ISurrogateSelector? SurrogateSelector { get; set; }
         [System.ObsoleteAttribute("BinaryFormatter serialization is obsolete and should not be used. See https://aka.ms/binaryformatter for more information.", DiagnosticId = "SYSLIB0011", UrlFormat = "https://aka.ms/dotnet-warnings/{0}")]
+        [System.Diagnostics.CodeAnalysis.RequiresUnreferencedCodeAttribute("BinaryFormatter serialization is not trim compatible because the Type of objects being serialized cannot be statically discovered.")]
         public abstract object Deserialize(System.IO.Stream serializationStream);
         protected virtual object? GetNext(out long objID) { throw null; }
         protected virtual long Schedule(object? obj) { throw null; }
         [System.ObsoleteAttribute("BinaryFormatter serialization is obsolete and should not be used. See https://aka.ms/binaryformatter for more information.", DiagnosticId = "SYSLIB0011", UrlFormat = "https://aka.ms/dotnet-warnings/{0}")]
+        [System.Diagnostics.CodeAnalysis.RequiresUnreferencedCodeAttribute("BinaryFormatter serialization is not trim compatible because the Type of objects being serialized cannot be statically discovered.")]
```

Comment:
With Serialize/Deserialize being incompatible with trimming, there's really nothing meaningful you can do with System.Runtime.Serialization.Formatters.  Is it worth all this annotation throughout the assembly for it, especially since it's now on a path to obsoletion?  Can we just slap an attribute on the whole assembly that warns on usage of anything in the assembly?  Or am I misunderstanding how this all works?

Reply:
1. There are other APIs in this assembly that can still work, for example most of `FormatterServices`. Are all the APIs here going away? Or just `IFormatter` and friends?

2. `RequiresUnreferencedCodeAttribute` is only allowed on Methods and Ctors today. And there is even a debate on allowing it on classes. See https://github.com/mono/linker/issues/1742. So it can't be applied to an assembly, today.

Reply:
> There are other APIs in this assembly that can still work, for example most of FormatterServices. Are all the APIs here going away? Or just IFormatter and friends?

https://github.com/dotnet/designs/blob/main/accepted/2020/better-obsoletion/binaryformatter-obsoletion.md#entirety-of-binaryformatter-type-obsolete-as-warning